src/python/pants/pantsd/process_manager.py

```diff
@@ -448,32 +456,43 @@ def daemonize(self, pre_fork_opts=None, post_fork_parent_opts=None, post_fork_ch
     daemons. Having a disparate umask from pre-vs-post fork causes files written in each phase to
     differ in their permissions without good reason - in this case, we want to inherit the umask.
     """
+    fork_context = fork_context or self._noop_fork_context
+
+    def double_fork():
+      logger.debug('forking %s', self)
+      pid = os.fork()
+      if pid == 0:
+        os.setsid()
+        second_pid = os.fork()
+        if second_pid == 0:
+          return False
+        else:
+          if write_pid: self.write_pid(second_pid)
+          return True
+      else:
+        # This prevents un-reaped, throw-away parent processes from lingering in the process table.
+        os.waitpid(pid, 0)
+      return None
+
+    # Perform the double fork under the fork_context. Three outcomes are possible after the double
+    # fork: we're either the original process, the double-fork parent, or the double-fork child.
+    # These are represented by parent_or_child being None, True, or False, respectively.
```

Comment:
Maybe this could use a tuple(is_original, is_parent) or something more enum-y, rather than a tri-state boolean?

```
if parent_or_child:
  ...
else:
  ...
```

doesn't read fantastically...

Reply:
I ache for actual enums... sigh.

Reply:
https://pypi.python.org/pypi/enum/